lib/upgradedb/mysql.2022062808.php

```diff
@@ -23,11 +23,12 @@
 
 $this->BeginTrans();
 
-if (!$this->ResourceExists('voip_cdr_type_uniqueid_ukey', LMSDB::RESOURCE_TYPE_CONSTRAINT)) {
+if ($this->ResourceExists('voip_cdr_type_uniqueid_ukey', LMSDB::RESOURCE_TYPE_INDEX)) {
```

Comment:
To akurat działa dla UNIQUE KEY! Sprawdzałem. Spójrz:

<pre>lmsdemo=&gt; SELECT constraint_name FROM information_schema.table_constraints
                                                WHERE table_catalog = &apos;lmsdemo&apos; AND constraint_name ILIKE &apos;%_ukey&apos;;
                    constraint_name                     
--------------------------------------------------------
 cash_importid_ukey
 customerassignments_customergroupid_ukey
 customercallassignments_customercallid_ukey
 customerconsents_ukey
 customercontactproperties_contactid_name_ukey
 dbinfo_keytype_ukey
 location_city_types_name_ukey
 netdevicemacs_mac_ukey
 netdevicemacs_netdevid_label_ukey
 netdevicetypes_name_ukey
 numberplanusers_userid_ukey
 rtticketwatchers_ticketid_ukey
 uiconfig_section_var_userid_divisionid_ukey
 userdivisions_userid_divisionid_ukey
 vlans_customerid_ukey
 vlans_netnodeid_ukey
 wh_products_barcode_ukey
 wh_products_items_item_name_ukey
 wh_products_items_macs_mac_ukey
 wh_products_items_macs_product_item_id_label_ukey
 voip_cdr_direction_uniqueid_ukey
 customerextids_customerid_extid_serviceproviderid_ukey
(22 rows)

</pre>

Reply:
Ja testuję to tak:
```
git fetch origin
sudo su - postgres -c 'dropdb lmstest';
git checkout origin/stable-24;
sudo su - postgres -c 'createdb -E UTF8 -O lmsdbuser lmstest';
sudo su - postgres -c 'psql lmstest -f /var/www/html/lms/doc/lms.pgsql';
git checkout origin/master;
php devel/upgradedb.php
```

dostaję:
upgradedb.php
(C) 2001-2022 LMS Developers
Using file /etc/lms/lms.ini as config.
DB schema version bumped to 2022062807
DB schema upgrade errors:
ALTER TABLE voip_cdr DROP CONSTRAINT voip_cdr_uniqueid_key
ERROR:  constraint "voip_cdr_uniqueid_key" of relation "voip_cdr" does not exist
ALTER TABLE voip_cdr ADD CONSTRAINT voip_cdr_type_uniqueid_ukey UNIQUE (type, uniqueid)
ERROR:  current transaction is aborted, commands ignored until end of transaction block
UPDATE dbinfo SET keyvalue = '2022062808' WHERE keytype = 'dbversion'
ERROR:  current transaction is aborted, commands ignored until end of transaction block

Reply:
Dobra inny sposób naprawienia tego problemu - a456d96d995692b8b565c7ea7add270aa11bc2bf